### PR TITLE
Fix rendering of Back link's `href` and `text` for falsy values

### LINKED ISSUE #5189: Restore behaviour of Back Link component for falsy `text` and `href` options
## What

Make `text` and `href` correctly render default values when a falsy value is passed to the component, and add some tests checking this for `undefined`/`null` and different kinds of falsy values `0`,`''`, `false`.

## Why

The use of the `default` filter means only `undefined` values are replaced by the default value provided to the filter. This means a `''` string as `text` would make the button render with no content, causing both visual and accessibility issues.

## Who needs to work on this

Developers

## Who needs to review this

Developers

## Done when

- [ ] Default values for `text` and `href` render when passed a 'falsy' value
- [ ] Tests have been added for `undefined`/`null` and different kinds of falsy values `0`,`''`, `false`

### PR DESCRIPTION
**Pull request by romaricpascal**
Reverts to the behaviour of 5.4.0, which ensured consistency across falsy values, and importantly that a 'Back' text was displayed in case of empty strings (and a '#' value for the `href` attribute, but there's less difference with a '').

Adds tests to ensure we keep an eye on the behaviour in future template changes, or at least make sure that any deviation is intentional (and tested).

Instead of reverting the commit, I used the `boolean` option from Nunjucks' `default` filter to keep the tidy up of the templates introduced in https://github.com/alphagov/govuk-frontend/commit/fc704d1efc0fc15a69ff84eb18050dd7ba7ce4b3.

Fixes #5189

### CODE REVIEW

File: packages/govuk-frontend/src/govuk/components/back-link/template.jsdom.test.js [hunk 1]
```diff
@@ -35,32 +35,74 @@ describe('back-link component', () => {
...
+      )
+
+      const $component = document.querySelector('.govuk-back-link')
+      expect($component).toHaveAttribute('href', '/home')
+    })
 
-    const $component = document.querySelector('.govuk-back-link')
-    expect($component).toHaveAttribute('href', '/home')
+    it.each(['', 0, false, null, undefined])(
+      'displays `Back` for `%s`',
```
Comment (querkmachine) [inline packages/govuk-frontend/src/govuk/components/back-link/template.jsdom.test.js:50]: I think this test is misnamed, as it appears to be testing whether `href` is `#` when a value is falsy or undefined, not the link text.

The test for the link text is on line 78.

**Review by querkmachine — APPROVED** [review]
One note about a test having the wrong name, but otherwise this seems to do what it sets out to do, and it has tests now! Hurray!

File: packages/govuk-frontend/src/govuk/components/back-link/template.jsdom.test.js (hunk 1 above)
Comment (romaricpascal) [reply packages/govuk-frontend/src/govuk/components/back-link/template.jsdom.test.js:50]: Whoops, good catch, I'll rename.
